File: lib/features/properties/domain/entities/property.dart
class Property {
  final String id;
  final String sellerId;
  final String title;
  final String description;
  final String location;
  final double price;
  final int bedrooms;
  final int bathrooms;
  final List<String> images;
  final bool verified;
  final DateTime createdAt;

  Property({
    required this.id,
    required this.sellerId,
    required this.title,
    required this.description,
    required this.location,
    required this.price,
    required this.bedrooms,
    required this.bathrooms,
    required this.images,
    required this.verified,
    required this.createdAt,
  });
}

File: backend/src/models/property.js
const mongoose = require('mongoose');

const propertySchema = new mongoose.Schema({
  title: { type: String, required: true, trim: true },
  description: { type: String, trim: true },
  price: { type: Number, required: true },
  location: {
    address: String,
    lat: Number,
    lng: Number,
  },
  agentId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  listingType: { type: String, enum: ['free', 'featured', 'premium'], default: 'free' },
  isFeatured: { type: Boolean, default: false },
  featuredUntil: { type: Date },
  boostLevel: { type: Number, min: 0, max: 3, default: 0 },
  boostStartDate: { type: Date },
  boostEndDate: { type: Date },
  boostStatus: { type: String, enum: ['none','pending','approved','rejected'], default: 'none' },
  mockPaymentStatus: { type: String, enum: ['pending','paid','failed'], default: 'pending' },
  isVerified: { type: Boolean, default: false },
  viewsCount: { type: Number, default: 0 },
  inquiriesCount: { type: Number, default: 0 },
  createdAt: { type: Date, default: Date.now },
  updatedAt: { type: Date, default: Date.now },
  suspended: { type: Boolean, default: false },
});

propertySchema.index({ agentId: 1 });
propertySchema.index({ 'location.lat': 1, 'location.lng': 1 });

propertySchema.pre('save', function (next) {
  this.updatedAt = Date.now();
  next();
});

module.exports = mongoose.model('Property', propertySchema);
